File: src/Application Framework/Structures/DataGridContractColumn.cs
using System.Data;
using System.Drawing;
using System.Windows.Forms;

namespace FSM
{
    public class DataGridContractColumn : DataGridLabelColumn
    {
        

        protected override void Paint(Graphics g, Rectangle bounds, CurrencyManager source, int rowNum, Brush backBrush, Brush foreBrush, bool alignToRight)
        {
            // Need to get row information from data
            DataRowView row = (DataRowView)source.List[rowNum];

            // check for row before we call it
            bool renewed = false;
            if (row.DataView.Table.Columns.Contains("Renewed"))
            {
                var switchExpr = Entity.Sys.Helper.MakeStringValid(row["Renewed"]);
                switch (switchExpr)
                {
                    case "Renewed":
                        {
                            renewed = true;
                            break;
                        }

                    case "Transfered":
                        {
                            renewed = true;
                            break;
                        }

                    default:
                        {
                            renewed = false;
                            break;
                        }
                }
            }

            bool contactMade = false;
            if (row.DataView.Table.Columns.Contains("ContactMade"))
            {
                contactMade = Entity.Sys.Helper.MakeBooleanValid(row["ContactMade"]);
            }

            if (!renewed & !contactMade)
            {
                backBrush = Brushes.Red;
            }
            else if (!renewed & contactMade)
            {
                backBrush = Brushes.Yellow;
            }
            else
            {
                backBrush = backBrush;
            }

            base.Paint(g, bounds, source, rowNum, backBrush, foreBrush, alignToRight);
        }

        
    }
}

File: src/Application Framework/Structures/DataGridContractColumn.vb
Public Class DataGridContractColumn : Inherits DataGridLabelColumn

#Region "Functions"

    Protected Overloads Overrides Sub Paint(ByVal g As System.Drawing.Graphics, ByVal bounds As System.Drawing.Rectangle,
                                            ByVal source As System.Windows.Forms.CurrencyManager, ByVal rowNum As Integer,
                                            ByVal backBrush As System.Drawing.Brush, ByVal foreBrush As System.Drawing.Brush,
                                            ByVal alignToRight As Boolean)

        'Need to get row information from data
        Dim row As DataRow = [source].List.Item(rowNum).row

        'check for row before we call it
        Dim renewed As Boolean = False
        If row.Table.Columns.Contains("Renewed") Then
            Select Case Entity.Sys.Helper.MakeStringValid(row("Renewed"))
                Case "Renewed"
                    renewed = True
                Case "Transfered"
                    renewed = True
                Case Else
                    renewed = False
            End Select
        End If

        Dim contactMade As Boolean = False
        If row.Table.Columns.Contains("ContactMade") Then
            contactMade = Entity.Sys.Helper.MakeBooleanValid(row("ContactMade"))
        End If

        If Not renewed And Not contactMade Then
            backBrush = Brushes.Red
        ElseIf Not renewed And contactMade Then
            backBrush = Brushes.Yellow
        Else : backBrush = backBrush
        End If

        MyBase.Paint(g, bounds, source, rowNum, backBrush, foreBrush, alignToRight)

    End Sub

#End Region

End Class
